### PR TITLE
Follow-ups to #22587 and #23269

### PR DESCRIPTION
**Pull request by topimiettinen**
Add links to documentation to NetLabel, LSM, Netlabel Fallback Peer Labeling
and NFT.

Use a positive boolean function nft_identifier_good() to help grok the checks
more easily than nft_identifier_bad().

Escape logged invalid items.

Add an ASSERT_PTR() for 'data' in config_parse_netlabel().

Use a different debug message when NFT operation fails.

Align nfproto_table strings.

Remove NFT_SET_MSGS, it's only used once.

### CODE REVIEW

**Comment by topimiettinen** [comment]
focal-amd64 fails with
```
upstream             FAIL non-zero exit status 1
```
Probably not related since other focals and everything else pass the tests.

File: src/core/dbus-cgroup.c [hunk 1]
```diff
@@ -2101,10 +2101,10 @@ int bus_cgroup_set_property(
                         if (!nfproto_name)
                                 return sd_bus_error_setf(error, SD_BUS_ERROR_INVALID_ARGS, "Invalid protocol %d.", nfproto);
 
-                        if (nft_identifier_bad(table))
+                        if (!nft_identifier_good(table))
                                 return sd_bus_error_setf(error, SD_BUS_ERROR_INVALID_ARGS, "Invalid NFT table name %s.", table);
```
Comment (yuwata) [inline src/core/dbus-cgroup.c:2105]: Please also escape the string in error message.

File: src/core/dbus-cgroup.c [hunk 2]
```diff
@@ -2101,10 +2101,10 @@ int bus_cgroup_set_property(
...
 
-                        if (nft_identifier_bad(set))
+                        if (!nft_identifier_good(set))
                                 return sd_bus_error_setf(error, SD_BUS_ERROR_INVALID_ARGS, "Invalid NFT set name %s.", set);
```
Comment (yuwata) [inline src/core/dbus-cgroup.c:2108]: Also here.

File: src/core/dbus-execute.c [hunk 3]
```diff
@@ -3552,10 +3552,10 @@ int bus_exec_context_set_transient_property(
                         if (!nfproto_name)
                                 return sd_bus_error_setf(error, SD_BUS_ERROR_INVALID_ARGS, "Invalid protocol %d.", nfproto);
 
-                        if (nft_identifier_bad(table))
+                        if (!nft_identifier_good(table))
                                 return sd_bus_error_setf(error, SD_BUS_ERROR_INVALID_ARGS, "Invalid NFT table name %s.", table);
```
Comment (yuwata) [inline src/core/dbus-execute.c:3556]: Also here.

File: src/core/dbus-execute.c [hunk 4]
```diff
@@ -3552,10 +3552,10 @@ int bus_exec_context_set_transient_property(
...
 
-                        if (nft_identifier_bad(set))
+                        if (!nft_identifier_good(set))
                                 return sd_bus_error_setf(error, SD_BUS_ERROR_INVALID_ARGS, "Invalid NFT set name %s.", set);
```
Comment (yuwata) [inline src/core/dbus-execute.c:3559]: Also here.

File: src/network/networkd-netlabel.c [hunk 5]
```diff
@@ -150,8 +151,8 @@ int config_parse_netlabel(
                 const char *rvalue,
                 void *data,
                 void *userdata) {
+        Set **set = ASSERT_PTR(data);
```
Comment (yuwata) [inline src/network/networkd-netlabel.c:154]: Please insert a empty line between the function prototype and variable declaration.

Also, you can drop `assert(set);` below.

File: src/shared/firewall-util-nft.c [hunk 6]
```diff
@@ -928,10 +929,16 @@ static int nft_set_element_op_in_addr_open(
         r = nft_set_element_op_in_addr(nfnl, table, set,
                                        add, nfproto, af, address, prefixlen);
 
-        log_debug("%s NFT family %s table %s set %s IP address %s",
-                  add ? "Added" : "Deleted",
-                  nfproto_to_string(nfproto), table, set,
-                  IN_ADDR_PREFIX_TO_STRING(af, address, prefixlen));
+        if (r < 0)
+                log_debug_errno(r, "Failed to %s NFT family %s table %s set %s IP address %s: %m",
```
Comment (yuwata) [inline src/shared/firewall-util-nft.c:933]: Maybe, better to add comma or colon between entries? E.g.
```
"Failed to %s NFT: family=%s, table=%s, set=%s, address=%s: %m"
```

File: src/shared/firewall-util-nft.c [hunk 7]
```diff
@@ -928,10 +929,16 @@ static int nft_set_element_op_in_addr_open(
...
+                                add ? "add" : "delete",
+                                nfproto_to_string(nfproto), table, set,
+                                IN_ADDR_PREFIX_TO_STRING(af, address, prefixlen));
+        else
+                log_debug("%s NFT family %s table %s set %s IP address %s",
```
Comment (yuwata) [inline src/shared/firewall-util-nft.c:938]: Also here.
```
"%s NFT: family=%s, table=%s, set=%s, address=%s"
```

File: src/shared/firewall-util-nft.c [hunk 8]
```diff
@@ -1137,21 +1144,19 @@ int fw_nftables_add_local_dnat(
...
-        [NFPROTO_ARP] = "arp",
+        [NFPROTO_ARP] =    "arp",
         [NFPROTO_BRIDGE] = "bridge",
-        [NFPROTO_INET] = "inet",
-        [NFPROTO_IPV4] = "ip",
-        [NFPROTO_IPV6] = "ip6",
+        [NFPROTO_INET] =   "inet",
+        [NFPROTO_IPV4] =   "ip",
+        [NFPROTO_IPV6] =   "ip6",
         [NFPROTO_NETDEV] = "netdev",
```
Comment (yuwata) [inline src/shared/firewall-util-nft.c:1152]: Please align with "=".

File: src/core/load-fragment.c [hunk 9]
```diff
@@ -6571,17 +6571,25 @@ static int config_parse_nft_set(
                         return 0;
                 }
 
-                if (nft_identifier_bad(table_resolved))
-                        return log_syntax(unit, LOG_WARNING, filename, line, 0, "Invalid table name %s, ignoring", table);
+                if (!nft_identifier_good(table_resolved)) {
+                        _cleanup_free_ char *esc = NULL;
+
+                        esc = cescape(table);
+                        return log_syntax(unit, LOG_WARNING, filename, line, 0, "Invalid table name %s, ignoring", esc);
```
Comment (yuwata) [inline src/core/load-fragment.c:6578]: `cescape()` may fail. Please wrap `esc` with `strna()`.

File: src/core/load-fragment.c [hunk 10]
```diff
@@ -6571,17 +6571,25 @@ static int config_parse_nft_set(
...
                         return 0;
                 }
 
-                if (nft_identifier_bad(set_resolved))
-                        return log_syntax(unit, LOG_WARNING, filename, line, 0, "Invalid set name %s, ignoring", set);
+                if (!nft_identifier_good(set_resolved)) {
+                        _cleanup_free_ char *esc = NULL;
+
+                        esc = cescape(set);
+                        return log_syntax(unit, LOG_WARNING, filename, line, 0, "Invalid set name %s, ignoring", esc);
```
Comment (yuwata) [inline src/core/load-fragment.c:6591]: Also here.

File: src/network/networkd-netlabel.c [hunk 11]
```diff
@@ -179,8 +180,11 @@ int config_parse_netlabel(
 
                 /* Label semantics depend on LSM but let's do basic checks */
                 if (!string_is_safe(w)) {
+                        _cleanup_free_ char *esc = NULL;
+
+                        esc = cescape(w);
                         log_syntax(unit, LOG_WARNING, filename, line, 0,
-                                   "Bad NetLabel label, ignoring: %s", w);
+                                   "Bad NetLabel label, ignoring: %s", esc);
```
Comment (yuwata) [inline src/network/networkd-netlabel.c:187]: Also here.

File: src/shared/firewall-util-nft.c [hunk 12]
```diff
@@ -1332,11 +1337,19 @@ int config_parse_nft_set_context(
...
 
-                if (nft_identifier_bad(table))
-                        return log_syntax(unit, LOG_WARNING, filename, line, 0, "Invalid table name %s, ignoring", table);
+                if (!nft_identifier_good(table)) {
+                        _cleanup_free_ char *esc = NULL;
 
-                if (nft_identifier_bad(set))
-                        return log_syntax(unit, LOG_WARNING, filename, line, 0, "Invalid set name %s, ignoring", set);
+                        esc = cescape(table);
+                        return log_syntax(unit, LOG_WARNING, filename, line, 0, "Invalid table name %s, ignoring", esc);
```
Comment (yuwata) [inline src/shared/firewall-util-nft.c:1344]: Also here.

File: src/shared/firewall-util-nft.c [hunk 13]
```diff
@@ -1332,11 +1337,19 @@ int config_parse_nft_set_context(
...
+                }
+
+                if (!nft_identifier_good(set)) {
+                        _cleanup_free_ char *esc = NULL;
+
+                        esc = cescape(set);
+                        return log_syntax(unit, LOG_WARNING, filename, line, 0, "Invalid set name %s, ignoring", esc);
```
Comment (yuwata) [inline src/shared/firewall-util-nft.c:1351]: Also here.

**Review by yuwata — CHANGES_REQUESTED** [review]
Please do not squash changes. At least, please split the changes into two: for NFT and for NetLabel.

**Comment by yuwata** [comment]
Now I am preparing a PR based on this. Please wait for a while.

**Comment by yuwata** [comment]
Sorry, #22587 and #23269 are reverted. Please resubmit them with this change. Closing.

**Pull request closed without merging**